onnx/defs/function.cc

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) Facebook Inc. and Microsoft Corporation.
+// Licensed under the MIT license.
+
+#include "onnx/defs/function.h"
+#include "onnx/checker.h"
+
+namespace ONNX_NAMESPACE {
+using namespace checker;
+FunctionBuilder& FunctionBuilder::SetDomain(const std::string& domain) {
+  domain_ = domain;
+  return *this;
+}
+
+const std::string& FunctionBuilder::GetDomain() const {
+  return domain_;
+}
+
+FunctionBuilder& FunctionBuilder::SetBuildFunction(BuildFunction build_func) {
+  build_func_ = build_func;
+  return *this;
+}
+
+BuildFunction FunctionBuilder::GetBuildFunction() const {
+  return build_func_;
+}
+
+Status FunctionBuilderRegistry::Register(
+    const FunctionBuilder& function_builder) {
+  std::lock_guard<std::mutex> lock(mutex_);
+  function_builders.push_back(function_builder);
+  return Status::OK();
+}
+
+// Get functions for specific domain.
+Status FunctionBuilderRegistry::GetFunctions(
+    const std::string& domain,
+    /*out*/
+    std::multimap<std::string, std::shared_ptr<FunctionProto>>* function_set)
+    const {
+  if (nullptr == function_set) {
```

Comment:
this seems to be worth to create a macro/function in status.{h,cc} because there will be quite often such check.

Reply:
You mean macro to create status, right? given I'm using the MakeString in string_utils, I feel it's not needed. the macro will look similar as the current lines of codes.